Source/WebCore/css/DOMCSSNamespace.cpp

```diff
@@ -77,17 +64,8 @@ bool DOMCSSNamespace::supports(Document& document, const String& property, const
     if (propertyID == CSSPropertyInvalid)
         return false;
 
-    // CSSParser::parseValue() won't work correctly if !important is present,
-    // so just get rid of it. It doesn't matter to supports() if it's actually
-    // there or not, provided how it's specified in the value is correct.
-    String normalizedValue = value.stripWhiteSpace().simplifyWhiteSpace();
-    normalizedValue = valueWithoutImportant(normalizedValue);
-
-    if (normalizedValue.isEmpty())
-        return false;
-
```

Comment:
nice code deletion!

Reply:
Taking out the empty string check wasn’t correct. Now we hit an assertion in `CSSParser::parseValue` if the passed in `value` is empty. We need to restore this:
```
if (value.isEmpty())
    return false;
```
Or alternatively we can change `CSSParser::parseValue` to support the empty string.

Reply:
Let me ask Webkit Bot to revert this and then I will keep it.